src/Serilog/Configuration/LoggerSinkConfiguration.cs

```diff
@@ -128,9 +128,21 @@ internal LoggerSinkConfiguration(LoggerConfiguration loggerConfiguration, Action
             var lc = new LoggerConfiguration();
 
             _applyInheritedConfiguration(lc);
-
             configureLogger(lc);
-            return Sink(new SecondaryLoggerSink(lc.CreateLogger(), attemptDispose: true), restrictedToMinimumLevel, levelSwitch);
+
+            var subLogger = lc.CreateLogger();
+            if (subLogger.HasOverrideMap)
+            {
+                // Overrides in sub-loggers are not supported and will be ignored
+                SelfLog.WriteLine("We have detected that you are using .MinimumLevel.Override() from a sub-logger (.WriteTo.Logger(Action<LoggerConfiguration>)). " +
+                                  "Overrides in a sub-logger are not supported and are therefore ignored. " +
+                                  "We recommend applying .MinimulLevel.Override on the root logger or " +
+                                  "using .Filter in your sub-loggers. " +
+                                  "Note that support for sub-loggers' minimum level overrides may be removed in the next major version.");
+            }
+
+            ILogEventSink secondaryLoggerSink = new SecondaryLoggerSink(subLogger, attemptDispose: true);
```

Comment:
Can use `var`. Variable with the same purpose is called `secondarySink` in the method below.